src/main/src/test/java/org/geoserver/ows/URLManglersTest.java

```diff
@@ -55,4 +71,18 @@ public void testProxyBase() {
         String url = buildURL(BASEURL, "test", null, URLType.SERVICE);
         assertEquals("http://geoserver.org/test", url);
     }
+
+    @Test
+    public void testProxyBaseParametrized() {
+        GeoServerInfo gi = getGeoServer().getGlobal();
+        gi.getSettings().setProxyBaseUrl("${proxy.custom}");
+        getGeoServer().save(gi);
+
+        String url = buildURL(BASEURL, "test", null, URLType.SERVICE);
+        if (GeoServerEnvironment.ALLOW_ENV_PARAMETRIZATION) {
```

Comment:
Looks like there is no way to activate ALLOW_ENV_PARAMETRIZATION constant by test class basis on maven builds currently due to the design of the feature, an improvement on it will be required (maybe replace it with an static method with status change mechanism available?).

Reply:
The variable could be made non final?

Reply:
Well, I think it could be made non final, it can't become null since is a primitive boolean type so no danger of a NPE.  The only concern I have is the variable will be available as public for modification from every part of the code including extensions and on every thread (should be volatile as well?), so not the best fix for the issue but could work as a fast workaround.

Reply:
If this was GeoToools, it would be pretty bad indeed. But GeoServer is an application, and the code is not really meant for "reuse" outside its confines. Probably a better way is to encapsulate the variable in a method call (it's gonna look better anyways, a quick refactor away, and it's not an extension point anyways, we have lower consideration of public facing API in GeoServer). Then add a synchronized method for modifying it, and warn in javadoc that it's not meant to be used outside testing.

Be careful of extension points and community modules using it (few of them):

```
src/community/backup-restore/extension/src/main/java/org/geoserver/backuprestore/imagemosaic/reader/ImageMosaicAdditionalResourceReader.java:import org.geoserver.platform.GeoServerEnvironment;
src/community/backup-restore/extension/src/main/java/org/geoserver/backuprestore/imagemosaic/reader/ImageMosaicAdditionalResourceReader.java:    final GeoServerEnvironment gsEnvironment = GeoServerExtensions.bean(GeoServerEnvironment.clas
s);
src/community/backup-restore/extension/src/main/java/org/geoserver/backuprestore/imagemosaic/reader/ImageMosaicAdditionalResourceReader.java:            if (GeoServerEnvironment.ALLOW_ENV_PARAMETRIZATION) {
src/community/security/keycloak/src/main/java/org/geoserver/security/keycloak/GeoServerKeycloakFilterConfig.java:import org.geoserver.platform.GeoServerEnvironment;
src/community/security/keycloak/src/main/java/org/geoserver/security/keycloak/GeoServerKeycloakFilterConfig.java:        final GeoServerEnvironment gsEnvironment =
src/community/security/keycloak/src/main/java/org/geoserver/security/keycloak/GeoServerKeycloakFilterConfig.java:                GeoServerExtensions.bean(GeoServerEnvironment.class);
src/community/security/keycloak/src/main/java/org/geoserver/security/keycloak/GeoServerKeycloakFilterConfig.java:                && GeoServerEnvironment.ALLOW_ENV_PARAMETRIZATION) {
src/extension/authkey/src/main/java/org/geoserver/security/AuthenticationKeyFilterConfig.java:import org.geoserver.platform.GeoServerEnvironment;
src/extension/authkey/src/main/java/org/geoserver/security/AuthenticationKeyFilterConfig.java:            final GeoServerEnvironment gsEnvironment =
src/extension/authkey/src/main/java/org/geoserver/security/AuthenticationKeyFilterConfig.java:                    GeoServerExtensions.bean(GeoServerEnvironment.class);
src/extension/authkey/src/main/java/org/geoserver/security/AuthenticationKeyFilterConfig.java:                                && GeoServerEnvironment.ALLOW_ENV_PARAMETRIZATION) {
src/extension/authkey/src/test/java/org/geoserver/security/AuthKeyAuthenticationTest.java:import org.geoserver.platform.GeoServerEnvironment;
src/extension/authkey/src/test/java/org/geoserver/security/AuthKeyAuthenticationTest.java:        Field field = GeoServerEnvironment.class.getDeclaredField("ALLOW_ENV_PARAMETRIZATION");
src/extension/authkey/src/test/java/org/geoserver/security/AuthKeyAuthenticationTest.java:        Field field = GeoServerEnvironment.class.getDeclaredField("ALLOW_ENV_PARAMETRIZATION");
src/extension/authkey/src/test/java/org/geoserver/security/AuthKeyAuthenticationTest.java:        final GeoServerEnvironment gsEnvironment =
src/extension/authkey/src/test/java/org/geoserver/security/AuthKeyAuthenticationTest.java:  
```